### PR TITLE
organization data

### PR DESCRIPTION
**Pull request by piyushtiw**

### CODE REVIEW

**Comment by piyushtiw** [comment]
@sudeeptarlekar _a please review it.

**Comment by vipulnsward** [comment]
@breezyboa _a

File: app/controllers/api/v1/organization_controller.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+module Api
+  module V1
+    class OrganizationController < BaseController
+      def data
+        org = Organization.find(params[:organization_id])
```
Comment (sudeeptarlekar) [inline app/controllers/api/v1/organization_controller.rb]: use `!` version here

File: test/graphql/resolvers/organization_serach_test.rb [hunk 2]
```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
```
Comment (sudeeptarlekar) [inline test/graphql/resolvers/organization_serach_test.rb]: correct file name here

File: app/controllers/api/v1/organization_controller.rb (hunk 1 above)
Comment (piyushtiw) [reply app/controllers/api/v1/organization_controller.rb]: I think No need to use '!' operator with find function.

File: app/javascript/Admin/Data/OrganizationData.elm [hunk 3]
```diff
@@ -0,0 +1,35 @@
+module Data.OrganizationData exposing (..)
```
Comment (breezyboa) [inline app/javascript/Admin/Data/OrganizationData.elm]: Please name this Data.Organization

File: app/javascript/Admin/Main.elm [hunk 4]
```diff
@@ -10,6 +10,9 @@ import Page.Url.List as UrlList
 import Page.Url.Create as UrlCreate
 import Page.Category.List as CategoryList
 import Page.Category.Create as CategoryCreate
+import Page.Organization.Show as OrganizationShow
```
Comment (breezyboa) [inline app/javascript/Admin/Main.elm]: OrganizationShow does not sound accurate. Please give this a more appropriate name

File: app/javascript/Admin/Main.elm [hunk 5]
```diff
@@ -213,10 +230,49 @@ retrivePage pathname =
...
+                ( OrganizationShow pageModel, Cmd.map OrganizationShowMsg pageCmd )
+
         _ ->
             ( NotFound, Cmd.none )
 
 
+extractStaticPath : Location -> String
+extractStaticPath location =
+        let
+            staticPath = (Debug.log "" (parsePath ( s "admin" </> s "organization" </> string ) location))
```
Comment (breezyboa) [inline app/javascript/Admin/Main.elm]: Remove Debug.log

File: app/javascript/Admin/Main.elm [hunk 6]
```diff
@@ -213,10 +230,49 @@ retrivePage pathname =
...
+                    OrganizationShow.init (retriveOrganizationFromUrl location)
+            in
+                ( OrganizationShow pageModel, Cmd.map OrganizationShowMsg pageCmd )
+
         _ ->
             ( NotFound, Cmd.none )
 
 
+extractStaticPath : Location -> String
+extractStaticPath location =
```
Comment (breezyboa) [inline app/javascript/Admin/Main.elm]: Whats the purpose of this function looks like its not doing anything?
its only being used in retrivePage and there too cant you just do location.pathname?

File: app/javascript/Admin/Main.elm [hunk 7]
```diff
@@ -213,10 +230,49 @@ retrivePage pathname =
...
+                        "/admin/organization"
+
+
+
+retriveOrganizationFromUrl : Location -> OrganizationId
+retriveOrganizationFromUrl location =
+   let
+      org = parsePath ( s "admin" </> s "organization" </> int ) location
+   in
+       getOrganizationId ( org )
```
Comment (breezyboa) [inline app/javascript/Admin/Main.elm]: What happens if organization id is -1? what would the user see?
I assume this is an error condition wherein organization was not found an appropriate message needs to be displayed to the user

File: app/javascript/Admin/Page/Organization/Show.elm [hunk 8]
```diff
@@ -0,0 +1,94 @@
+module Page.Organization.Show exposing (..)
```
Comment (breezyboa) [inline app/javascript/Admin/Page/Organization/Show.elm]: Show does not seem like an appropriate name.

File: app/javascript/Admin/Page/Organization/Show.elm [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+renderArticles articles =
+     List.map renderArticle articles
+
+
+
+fetchOrganization : OrganizationId -> Cmd Msg
+fetchOrganization organiztionId =
+    let
+        request =
+            requestOrganization "dev" "3c60b69a34f8cdfc76a0" organiztionId
```
Comment (breezyboa) [inline app/javascript/Admin/Page/Organization/Show.elm]: "dev" should not be hardcoded. Actual environment should be passed in

**Comment by breezyboa** [comment]
@piyushtiw _a

File: app/javascript/Admin/Main.elm (hunk 6 above)
Comment (piyushtiw) [reply app/javascript/Admin/Main.elm]: It is redirecting "/admin/organization/:dynamic_id" url to OrganizationDisplay page.

**Comment by piyushtiw** [comment]
@breezyboa _a

File: app/javascript/Admin/Main.elm (hunk 6 above)
Comment (breezyboa) [reply app/javascript/Admin/Main.elm]: My question is if there is no use for this dynamic id why have it in the first place and introduce this complexity? why not always goto the static path

**Pull request merged**